src/sentry/tasks/merge.py

```diff
@@ -52,14 +72,18 @@ def merge_group(
         GroupMeta,
     )
 
-    if not (from_object_id and to_object_id):
+    if not (from_object_ids and to_object_id):
         logger.error(
             'group.malformed.missing_params', extra={
                 'transaction_id': transaction_id,
             }
         )
         return
 
+    # Operate on one "from" group per task iteration. The task is recursed
+    # until each group has been merged.
+    from_object_id = from_object_ids[0]
+
     try:
         group = Group.objects.get(id=from_object_id)
```

Comment:
One group probably shouldn't prevent the rest of the groups from being attempted to be merged, I think?

Similarly (noting it here, because I can't add an inline comment on unchanged blocks) should L100 also handle group redirects?

Reply:
Damb good catch.

RE: group redirects do you mean moving the `GroupRedirect` creation up before the rest of the merge is performed?

https://github.com/getsentry/sentry/blob/b94d3035a398d3a784d2ac02ba37916362e4dd42/src/sentry/tasks/merge.py#L172-L179

Or something else?

Reply:
Yeah — sorry, I re-read that and realized it wasn't super clear. What I'm thinking is there are two tasks:

1. Groups A, B, and C are merged into Group X.
2. Group X is merged into Group Y.

If these tasks occur out of order (so 2 prior to 1, because computers), when task 2 is completed, it's going to delete group X, and the first task will fail and one of the following will happen:

1. No groups get merged into group X.
2. Group A will end up merged into group X (and then subsequently Y.)
3. Group A and B will end up merged into group X (and then subsequently Y.)
4. Group A, B and C will end up merged into group X (and then subsequently Y.)

We can use the `GroupRedirect` model to ensure that all of A, B, and C are eventually merged into one of X or Y, since it'll have a record that shows that the group that used to be X is now Y, even after X is deleted (as long as we don't end up checking it between when X is deleted and the `GroupRedirect` row is written.)